### ISSUE TITLE
Combat Procs firing whilst a perk action is being 'used' results in broken NPC logic.

### ISSUE THREAD

**Issue by RaggyLP** [project-rubika/issues#646]
**Describe the bug**
If a combat-based proc such as shade proc nanos, or perk-related procs such as Accumulator fire whilst a perk action is being used (Special bar in top left), this results in NPCs fight logic totally breaking, causing them to become either unhittableable (all attacks miss) whilst still attacking, or unresponsive entirely (no auto attack, no player tracking).

**To Reproduce**
1. Grab a Shade and cast a proc nano, one with high % such as ```.ncu run 224167```
2. Have a set of spammable perks such as the Spirit Phylactery line.
3. Use ```.set debug 1``` on the mob and yourself to track fight state and proc rolls.
4. Start combat and begin to cycle the Perk lines whilst auto-attacking.
5. If the proc hits during the 'Special' bar of a perk cast, it will result in the mob becoming completely inert or unhittable whilst attacking.

**Expected behavior**
Overlapping perks/procs shouldn't cause a break in the NPC Fighting brain.

**Screenshots**
4. / 5.
![image](https://github.com/project-rubika/issues/assets/156821187/6ae989e8-24bb-418b-9f7f-c290bf8687ef)

5. 
![image](https://github.com/project-rubika/issues/assets/156821187/72af46a3-0c0e-4cbf-b420-50e75267cfba)
![image](https://github.com/project-rubika/issues/assets/156821187/0ce2a3a3-fadb-47a1-ae45-3d96cd54d9b3)

**Comment by portalerprk** [comment]
Fix dropped. Test it out when ready.

This one is particularly insidious.

We've struggled at times understanding how targeting really works. For example the "CastNano" function on perks/item-procs/nanos/etc. Originally it was always the original caster that would execute all functions, with each function's target resolved in relation to the original caster. Then at some point we pivoted to functions being executed *inside the target's NCU*. This was because in some cases function target `self` was not in fact the caster but the target. Meaning the overall spell must have been executed from a new 'context'. 

Its very confusing.

So TLDR; I made a change that *might* fix this issue but may have unforeseen side affects elsewhere. I'm nervous about it.
